File: go/LeetCode/corpFlightBookings.go
package main

func main() {
	
}

func corpFlightBookings(bookings [][]int, n int) []int {
	res := make([]int, n)
	for _,v :=range bookings{
		res[v[0]-1]+=v[2]
		if v[1]<n {
			res[v[1]]-=v[2]
		}
	}
	for i:=1;i<n;i++ {
		res[i] +=res[i-1]
	}
	return res
}

File: python3/LeetCode/corpFlightBookings.py
from typing import List


class Solution:
    def corpFlightBookings(self, bookings:List[List[int]],n:int) -> List[int]:
        res = [0]*n
        for i in range (0, len(bookings)):
            res[bookings[i][0]-1]+=bookings[i][2]
            if bookings[i][1]<n:
                res[bookings[i][1]]-=bookings[i][2]

        for i in range (1,n):
            res[i]+=res[i-1]
        return res